### DISCUSSION TITLE
[native filters] Set default value using Jinja/SQL statement

### DISCUSSION THREAD

**Discussion by xneg** [apache/superset#19867] [Ideas]
### Idea
Currently Superset allows us to set default value choosing exact value from underlying dataset:

<img width="303" alt="picture" src="https://user-images.githubusercontent.com/5748886/165480556-f08a85cf-a6aa-47f7-861d-ba372bf84972.png">

It would be great to use instead Jinja template or SQL query. This is required to set the default value according to current user properties. 

There is [a similar PR](https://github.com/apache/superset/issues/12324) regarding old filters but this functionality is also needed in native filters.

### Implementation proposal
Native filter settings including default values are stored in `json_metadata`. If we add template processor (like [this one](https://github.com/apache/superset/blob/f91f83d011b4c83576f5fff76d502f5616f2972d/superset/jinja_context.py#L476)) when loading json_metadata we could achieve this goal. 
And you can customize your own default values from Advanced/Json metadata in dashboard properties:
<img width="471" alt="image" src="https://user-images.githubusercontent.com/5748886/165483329-cf81a9f7-3ff1-4679-9f8d-fb224001f6bb.png">

Disadvantage of this approach is when editing filters from UI the values you have written in json_metadata will be overwritten.

**Comment by nodeworks** [comment]
Were you able to find a solution to this?
